index.jsx

```diff
@@ -119,26 +120,45 @@ export default {
         if (!action) {
             return;
         }
-        if (action.type === 'SERIAL_PORT_SELECTED') {
-            const { port } = action;
-            store.dispatch(FirmwareActions.validateFirmware(port.serialNumber, {
-                onValid: () => store.dispatch(RssiActions.open(port)),
-                onInvalid: () => store.dispatch({ type: 'FIRMWARE_DIALOG_SHOW', port }),
-            }));
-        }
-        if (action.type === 'SERIAL_PORT_DESELECTED') {
-            store.dispatch(RssiActions.close());
-        }
-        if (action.type === 'FIRMWARE_DIALOG_UPDATE_REQUESTED') {
-            const { port } = action;
-            store.dispatch(FirmwareActions.programFirmware(port.serialNumber, {
-                onSuccess: () => {
-                    store.dispatch(RssiActions.open(port));
-                    store.dispatch({ type: 'FIRMWARE_DIALOG_HIDE' });
-                },
-            }));
+
+        switch (action.type) {
+            case 'DEVICE_DESELECTED':
+                logger.info('Deselecting device');
+                store.dispatch(RssiActions.close());
+                break;
+
+            case 'DEVICE_SETUP_COMPLETE':
+                logger.info(`Selecting device with s/n ${action.device.serialNumber}`);
+                store.dispatch(RssiActions.open(action.device.serialport));
+                break;
```

Comment:
Could also listen for `DEVICE_SELECTED` and log something like "Validating formware for device with s/n".